### ISSUE TITLE
[0.50.01-SNAPSHOT] Thumper and Sniper weapons swapped with one another on various units and Mechs

### ISSUE THREAD

**Issue by Tzahr** [MegaMek/megamek#6056]
### Prerequisites and Pre-Issue Checklist

- [X] I'm reporting the issue to the correct repository:
- [MegaMek](https://github.com/MegaMek/megamek/issues)
- [MegaMekLab](https://github.com/MegaMek/megameklab/issues)
- [MekHQ](https://github.com/MegaMek/mekhq/issues)

- [X] I've tested the issue against at least the latest MILESTONE version
- [X] I've asked on the MegaMek Discord about the error
- [X] I've reviewed the BattleTech rules and MegaMek documentation, and I've confirmed that something isn't working as intended.
- [X] I've searched the Github tracker and haven't found the issue listed

### Severity *

Medium (Gameplay Limitation): Non-core functionality is impaired, providing a suboptimal but playable experience.

### Brief Description *

Various units that come equipped with Snipers or Thumpers appear as being armed with the other, while still carrying ammunition for the correct weapon. This so far appears to affect 'Mechs (Helepolis), vehicles, and field artillery teams. 

![image](https://github.com/user-attachments/assets/1ddc7a46-db53-43ac-b675-f34d255d5a7b)
![image](https://github.com/user-attachments/assets/032bb4be-4816-4eb6-ac3e-3e18377445f5)

Deploying those units will show them with a `NULL Weapon Name!` notation when hovered over. 
![image](https://github.com/user-attachments/assets/bae3e503-cf87-4c5c-96c1-f5fa9f15edc6)

### Operating System *

Windows 11

### Java Version *

21.0.3

### MegaMek Suite Version *

Free Text (type manually)

### Custom MegaMek Version

Nightly v0.50.1 - Rev. 72f857dfe36e0b762ea6736dd51112b924e73a11

### Attach Files

[megamek.log](https://github.com/user-attachments/files/17238461/megamek.log)

### Final Checklist

- [X] I've checked to make sure that this issue has not already been filed
- [X] I'm reporting only one issue in this ticket for clarity and focus

**Comment by Tzahr** [comment]
When loading these units in MML, they will load matching the unit card, which is often in wildly illegal. 
The Helepolis, which is ordinarily armed with one Sniper, for example, loads in armed with two Thumpers, and being 10t over weight.

**Comment by Tzahr** [comment]
As a test, I created a custom Mech that is armed with a **Sniper** artillery cannon.
It saves fine, and crucially, it loads correctly as a Sniper-armed 'Mech. 
Upon inspection of the .mtf file with Notepad++ I noticed that the .mtf claims all its slots are filled with `ISThumper`: 

> Weapons:1
> Sniper, Left Arm
> 
> Left Arm:
> Shoulder
> Upper Arm Actuator
> ISThumper
> ISThumper
> ISThumper
> ISThumper
> ISThumper
> ISThumper
> ISThumper
> ISThumper
> ISThumper
> ISThumper
> 
> Right Arm:
> Shoulder
> Upper Arm Actuator
> Lower Arm Actuator
> Hand Actuator
> -Empty-
> -Empty-
> -Empty-
> -Empty-
> -Empty-
> -Empty-
> -Empty-
> -Empty-
> 
> Left Torso:
> ISThumper
> ISThumper
> ISThumper
> ISThumper
> ISThumper
> ISThumper
> ISThumper
> ISThumper
> ISThumper
> ISThumper
> -Empty-
> -Empty-
> 
> Right Torso:
> ISSniperAmmo
> ISSniperAmmo
> ISSniperAmmo
> ISSniperAmmo
> -Empty-
> -Empty-
> -Empty-
> -Empty-
> -Empty-
> -Empty-
> -Empty-
> -Empty-

It appears that the Sniper weapon in .mtf is incorrectly listed as `ISThumper`. 
Likewise, in the .mtf for the Helepolis (included below for easy reference), the weapons are (correctly) listed as `ISSniper`, but when loaded become two Thumpers, likely because it interprets the `ISSniper `entries in the .mtf as being two Thumpers. 

Interestingly, this does not appear to affect Clan Thumpers or Snipers, because the Clan Field Artillery unit is not affected - likely because their data entries use a different part ID. 

It seems therefore that one of the loading/saving processes is mixing up `ISSniper` and `ISThumper`, which is causing the units to load with completely the wrong weapon. 

[mechfiles.zip](https://github.com/user-attachments/files/17238683/mechfiles.zip)
